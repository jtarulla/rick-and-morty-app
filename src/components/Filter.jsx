import React from 'react';

const Filter = ({handleChange}) => {
    const handleInputChange = (event) => {
        event.preventDefault();
        handleChange(event.target.value);
    };

    return (
        <div className="filter">
            <div className="container">
                <input
                    onChange={handleInputChange}
                    placeholder="filter by name"
                    data-testid="input-filter"
                />
            </div>
        </div>
    )
}

export default Filter;