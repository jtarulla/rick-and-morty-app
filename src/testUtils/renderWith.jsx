import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
    return (
        <MemoryRouter initialEntries={initialRoutes}>
            {children}
        </MemoryRouter>
    );
};

export const renderWithRouter = (ui, options) => {
    const initialRoutes =
        options && options.initialRoutes ? options.initialRoutes : ["/"];

    return render(ui, {
            wrapper: (args) =>
                MemoryRouterWithInitialRoutes({
                    ...args,
                    initialRoutes,
                }),
                ...options,
        }
    );
}

export * from "@testing-library/react";
