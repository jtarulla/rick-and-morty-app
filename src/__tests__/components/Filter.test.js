import '@testing-library/jest-dom/extend-expect';
import { renderWithRouter } from 'testUtils/renderWith';
import { fireEvent, screen } from '@testing-library/react';
import Filter from 'components/Filter';

const setup = async () => {
    const utils = await renderWithRouter(<Filter handleChange={jest.fn} />);

    const input = await screen.findByTestId('input-filter');

    return {
        input,
        ...utils,
    }
}

describe('Filter component', () => {
    it('renders', async () => {
        const { input } = await setup();

        expect(input).toMatchSnapshot();
    });

    it('changes input value', async () => {
        const { input } = await setup();

        fireEvent.change(input, {target: {value: 'rick'}})
        expect(input.value).toBe('rick')
    })
})