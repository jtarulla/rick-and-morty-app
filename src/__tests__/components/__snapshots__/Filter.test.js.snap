// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders 1`] = `
<input
  data-testid="input-filter"
  placeholder="filter by name"
/>
`;
